HerPublicWebsite.BusinessLogic/Services/ReferralFollowUpManager/ReferralFollowUpService.cs

```diff
@@ -0,0 +1,56 @@
+using HerPublicWebsite.BusinessLogic.ExternalServices.EmailSending;
+
+using HerPublicWebsite.BusinessLogic.Models;
+
+namespace HerPublicWebsite.BusinessLogic.Services.ReferralFollowUps;
+
+public interface IReferralFollowUpService
+{
+    public Task GenerateAndSendFollowUpEmail(ReferralRequest referralRequest);
+    public ReferralRequestFollowUp GetReferralRequestFollowUpByToken(string token);
+    public Task RecordFollowUpResponseForToken(string token, bool hasFollowedUp);
+}
+
+public class ReferralFollowUpService : IReferralFollowUpService
+{
+    private readonly IEmailSender emailSender;
+    private readonly IDataAccessProvider dataAccessProvider;
+    private readonly IGuidService guidService;
+    public ReferralFollowUpService(IEmailSender emailSender, IDataAccessProvider dataAccessProvider, IGuidService guidService)
+    {
+        this.emailSender = emailSender;
+        this.dataAccessProvider = dataAccessProvider;
+        this.guidService = guidService;
+    }
+
+    public async Task GenerateAndSendFollowUpEmail(ReferralRequest referralRequest)
+    {
+        string token = GenerateFollowUpToken();
+
+        string followUpLink = "https://localhost:5001/referral-follow-up/respond-page/" + token;
```

Comment:
I wanted to ask where we could pull the hostname for in the app. I assume there's a nice DotNet-ty way of doing this

Reply:
I've now added this as a config variable. I was a bit unsure on what standard naming conventions should be. I've added a specific configuration class for the ReferralFollowUpNotificationService. But in appsettings.json I've made the base url a key of "Global" settings, as it felt weird to make it specific to the notification service.

That being said it feels a bit strange to have the ConfigSection key used by the notification configuration class not match the name of the class. 

Keen to know if you have any preferences on this

Reply:
Commented on the file, but I think it makes sense to keep the key `Global`, but make the configuration class reflect that.

(This is what we've done in the equivalent portal code)